codes/Graph/2SAT.cpp

```diff
@@ -25,22 +23,19 @@ class TwoSat {
     add_edge(x ^ 1, y); add_edge(y ^ 1, x);
   }
   bool solve() {
-    vis.assign(n, false);
     for (int i = 0; i < n; ++i) if (not vis[i]) dfs(i);
     reverse(ord.begin(), ord.end());
-    for (int u : ord) {
-      if (!vis[u]) continue;
-      sccs.push_back(vector<int>()); rdfs(u);
-    }
+    for (int u : ord) if (vis[u])
+      sccs.emplace_back(), rdfs(u);
```

Comment:
I'm not familiar enough with the order of evaluation of comma operator?
Thus, just want to double-check that this is correct.

Reply:
[stackoverflow](https://stackoverflow.com/questions/54142/how-does-the-comma-operator-work/) says comma operator introduces sequence point. Also I just tested it on CSES and got accepted.